src/lib/index.ts

```diff
@@ -344,8 +344,15 @@ class ServerlessIamPerFunctionPlugin {
     const isInherit = functionObject.iamRoleStatementsInherit
       || (this.defaultInherit && functionObject.iamRoleStatementsInherit !== false);
 
-    if (isInherit && !_.isEmpty(this.serverless.service.provider.iamRoleStatements)) { // add global statements
-      for (const s of this.serverless.service.provider.iamRoleStatements) {
+    // Since serverless 2.24.0 provider.iamRoleStatements is deprecated
+    // https://github.com/serverless/serverless/blob/master/CHANGELOG.md#2240-2021-02-16
+    // Support old & new iam statements by checking if `iam` property exists
+    const providerIamRoleStatements = this.serverless.service.provider.iam
+      ? this.serverless.service.provider.iam.role?.statements
```

Comment:
[Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) has been added in [node v14](https://nodejs.org/uk/blog/release/v14.0.0/), but plugin requires  [nodev10](https://github.com/LironEr/serverless-iam-roles-per-function/blob/feat/support_new_iam_structure/package.json#L6) and moreover readme says `Note: Node 6.10 or higher runtime required.` I believe we should fix it to require the same version everywhere.